client-state/models/Form.ts

```diff
@@ -160,31 +156,20 @@ export const Form = types
         self.fields.sort((a, b) => a.order - b.order)
       })
     },
-    buildQueryStringWithFormData(): string {
+    buildObjectWithFormData(): { [key: string]: string } {
       const parent = getParent(self) as Instance<typeof RootStore>
-
-      let qs = `_language=${parent.lang}`
-
+      let input = { _language: parent.lang }
       for (const field of self.fields) {
         if (!field.value) continue
-
-        // remove masking from currency and use object keys for react-select
-        let val = field.sanitizeInput()
-
-        if (qs !== '') qs += '&'
-        //encodeURI and fix for encodeURIComponent and circle brackets
-        qs += `${field.key}=${fixedEncodeURIComponent(val)}`
+        input[field.key] = field.sanitizeInput()
       }
-      return qs
+      return input
     },
   }))
   .actions((self) => ({
     sendAPIRequest: flow(function* () {
-      // build query  string
-      const queryString = self.buildQueryStringWithFormData()
-
-      const apiData = yield fetch(`/${self.API_URL}?${queryString}`)
-      const data: ResponseSuccess | ResponseError = yield apiData.json()
+      const input = self.buildObjectWithFormData()
+      const data = mainProcessor(input)
```

Comment:
I don't know definitively if Curam is forever out of the picture or not. I do know that if it is needed, it is still possible to use the existing API and build off that, but the more likely solution would be a new endpoint anyways, which means this change shouldn't affect Curam at all. The only case where this would mess things up is if we needed to communicate with some external system (db or Curam) on every frontend request. I don't that will be the case anymore, and a Curam button would be a new endpoint that's triggered by a new button in the UI.